src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2443,6 +2443,18 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     GenerateImplicitConversionError(diagnostics, expression.Syntax, conversion, expression, targetType);
                 }
 
+                if (expression.Kind == BoundKind.TupleLiteral)
+                {
+                    // A BoundTupleLiteral must always be upgraded to a BoundConvertedTupleLiteral during conversion
+                    var sourceTuple = (BoundTupleLiteral)expression;
+                    expression = new BoundConvertedTupleLiteral(
+                        sourceTuple.Syntax,
+                        sourceTuple.Type,
+                        sourceTuple.Arguments,
+                        sourceTuple.Type ?? targetType,
```

Comment:
It's a catch-22. What is the converted type if there was no conversion?

There are two cases:
1. the tuple had no natural type. From discussion with @VSadov, I think I'll use error type instead of `targetType` for that case.
2. the tuple had a natural type. It seems reasonable to keep it, but I'm open to suggestions.